shed/event_streams.py

```diff
@@ -690,10 +692,11 @@ class filter(EventStream):
     full_event: bool, optional
```

Comment:
for `input_info`, let's be a little more verbose to help (we'd have to remember if we change it though). maybe something like:

describe the incoming streams in the form of `{'from' : 'to'}` or `{'from' : ('to', stream_number)}` pairs.

Reply:
ok, although I would like to use `{'function_arg/kwarg_position/name': (('keys', 'to', 'data', 'in', 'dict'), stream_number)}` Since that is a bit more realistic (also `from` and `to` may be rather confusing)

Reply:
sounds good. i did not realize you could specify `(tuple(multiple_keys), stream_number)`. thanks

Reply:
Yes, this is part of my plan to eliminate `full_event`. This way if you give a tuple of keys it will dig down into the document that you provide (so you can access things like timestamps or data inside dicts inside of events). One awesome side benefit is that we now can just hand full dicts over by not specifying all the way down to the data itself.